lib/meta.nix

```
@@ -92,12 +92,15 @@ rec {
 
      A package is available on a platform if both
 
-       1. One of `meta.platforms` pattern matches the given platform.
+       1. One of `meta.platforms` pattern matches the given
+          platform, or `meta.platforms` is not present.
 
        2. None of `meta.badPlatforms` pattern matches the given platform.
   */
   availableOn = platform: pkg:
-    lib.any (platformMatch platform) pkg.meta.platforms &&
+    if !(pkg?meta) then true else
+    (!(pkg.meta ? platforms) ||
+      lib.any (platformMatch platform) pkg.meta.platforms) &&
```

You can check the full attribute path using `?`.

```suggestion
    if !(pkg ? meta.platforms) then true else
    lib.any (platformMatch platform) pkg.meta.platforms &&
```

---

I think this altered the logic -- after the commit above `meta.badPlatforms` is ignored if `meta.platforms` is absent.  I have adjusted for this in 194a4ea1a236.